### PR TITLE
perf(DRM): pass `preferredKeySystems` to `filterManifest()`

### PR DESCRIPTION
**Pull request by tykus160**
Follow up to #5391
Always use `preferredKeySystems` during manifest filtering, not only via DRM Engine.

### CODE REVIEW

File: test/util/stream_utils_unit.js [hunk 1]
```diff
@@ -551,7 +551,8 @@ describe('StreamUtils', () => {
         });
       });
 
-      await shaka.util.StreamUtils.filterManifest(fakeDrmEngine, manifest);
+      await shaka.util.StreamUtils.filterManifest(fakeDrmEngine, manifest,
```
Comment (avelad) [inline test/util/stream_utils_unit.js:554]: If you put a default value in the function, it is not necessary to change these tests.

**Review by avelad — APPROVED** [review]

**Comment by joeyparrish** [comment]
I see the "perf" label... Do you have any rough data on how much of a performance improvement this is?

**Review by joeyparrish — APPROVED** [review]

**Comment by tykus160** [comment]
@joeyparrish previous PR reduced load i.e. on Xbox Series S from ~400 ms to ~100 ms as it avoided 10 calls to widevine. This one does relatively the same, but on manifest updates.

**Comment by avelad** [comment]
@tykus160 Since Xbox only supports PlayReady, I wonder if, given this improvement, we should add the default PlayReady `drm.preferredKeySystems` config for Xbox. What do you think?

**Comment by avelad** [comment]
ClearKey it's necessary, so, for me the platforms are Xbox and PlayStation since only supports PlayReady .

ClearKey support will eventually come to Safari, so I don't want to limit the platform.

**Pull request merged**